src/test/java/org/jabref/logic/pdf/search/indexing/PdfIndexerTest.java

```diff
@@ -0,0 +1,151 @@
+package org.jabref.logic.pdf.search.indexing;
+
+
+import java.io.IOException;
+import java.util.Optional;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.MultiFields;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PdfIndexerTest {
+
+    private PdfIndexer indexer;
+    private BibDatabase database;
+
+    @Before
+    public void setUp() throws IOException {
+        this.indexer = new PdfIndexer();
+        this.database = mock(BibDatabase.class);
```

Comment:
Same as above, you don't need to mock them
You only need to mock classes which operate on preferences